### PR TITLE
Dependency refresh

### PR DESCRIPTION
**Pull request by huntc**
Pretty dramatic dependency refresh including Scala 2.13. This will force us to make a v2.0.0 as it is binary-incompatible with Scala 2.12 given collection deprecation amendments.

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -1,7 +1,9 @@
...
 
-danglingParentheses               = true
-indentOperator                    = spray
+preset = "defaultWithAlign"
+
+danglingParentheses.preset        = true
+indentOperator.preset             = spray
 maxColumn                         = 100
 newlines.alwaysBeforeMultilineDef = true
 project.excludeFilters            = [".*\\.sbt"]
```
Comment (huntc) [inline .scalafmt.conf:9]: As per the latest opinions of sbt-fresh, which we've always abided by.

File: src/main/scala/au/com/titanclass/streams/telemetry/MetricsJsonProtocol.scala [hunk 2]
```diff
@@ -85,14 +85,14 @@ object MetricsJsonProtocol extends DefaultJsonProtocol {
...
             .toMap
         ),
-        "counters"   -> JsObject(obj._2.asScala.mapValues(_.toJson).toMap),
-        "histograms" -> JsObject(obj._3.asScala.mapValues(_.toJson).toMap),
-        "meters"     -> JsObject(obj._4.asScala.mapValues(_.toJson).toMap),
-        "timers"     -> JsObject(obj._5.asScala.mapValues(_.toJson).toMap)
+        "counters"   -> JsObject(obj._2.asScala.view.mapValues(_.toJson).toMap),
+        "histograms" -> JsObject(obj._3.asScala.view.mapValues(_.toJson).toMap),
+        "meters"     -> JsObject(obj._4.asScala.view.mapValues(_.toJson).toMap),
+        "timers"     -> JsObject(obj._5.asScala.view.mapValues(_.toJson).toMap)
```
Comment (huntc) [inline src/main/scala/au/com/titanclass/streams/telemetry/MetricsJsonProtocol.scala:95]: I've never actually bothered to understand Scala's views until now! Scala 2.13 makes them a first-class citizen.

File: src/test/scala/au/com/titanclass/streams/telemetry/TracingJsonProtocolTest.scala [hunk 3]
```diff
@@ -35,14 +34,11 @@ object TracingJsonProtocolTest extends TestSuite {
...
-    ActorMaterializer()
-
   implicit lazy val ec: ExecutionContext =
-    mat.executionContext
+    system.dispatcher
 
-  val tests = Tests {
-    'writeTracing - {
+  val tests: Tests = Tests {
+    test("writeTracing") {
```
Comment (huntc) [inline src/test/scala/au/com/titanclass/streams/telemetry/TracingJsonProtocolTest.scala:41]: This is the "new" utest way.

File: src/test/scala/au/com/titanclass/streams/telemetry/TracingJsonProtocolTest.scala [hunk 4]
```diff
@@ -66,7 +62,7 @@ object TracingJsonProtocolTest extends TestSuite {
         .map { s =>
           val json = s.toJson
           val re =
-            """\{"baggage":\[\],"duration":.*,"logs":\[\{"time":0,"fields":\{\},"message":"hello-world"\},\{"time":0,"fields":\{"f1":"1","f0":"0"\}\}\],"operationName":"some-span","references":\[\],"spanId":.*,"start":.*,"tags":\{"sampler.type":"const","sampler.param":"true"\},"traceId":.*\}""".r
+            """\{"baggage":\[],"duration":.*,"logs":\[\{"time":0,"fields":\{},"message":"hello-world"},\{"time":0,"fields":\{"f0":"0","f1":"1"}}],"operationName":"some-span","references":\[],"spanId":.*,"start":.*,"tags":\{"sampler.type":"const","sampler.param":"true"},"traceId":.*}""".r
           assertMatch(re.findFirstIn(json.compactPrint)) {
```
Comment (huntc) [inline src/test/scala/au/com/titanclass/streams/telemetry/TracingJsonProtocolTest.scala:66]: Its a bit crappy to make string comparisons given that differing dependencies will (have) affected the order. We're using regex here though, and so I don't want to over-complicate things.

File: src/main/scala/au/com/titanclass/streams/telemetry/MetricsJsonProtocol.scala (hunk 2 above)
Comment (stevewillcock) [reply src/main/scala/au/com/titanclass/streams/telemetry/MetricsJsonProtocol.scala:95]: Yes, I had the same thing the other day doing a groupBy on a Seq in 2.13 - deprecation notices all over the place until I swapped to use mapValues on the view

**Review by stevewillcock — APPROVED** [review]
Looks good, surprisingly few actual code changes to make it play nicely with 2.13

**Pull request merged**